hts.c

```diff
@@ -2153,7 +2153,7 @@ static int idx_read_core(hts_idx_t *idx, BGZF *fp, int fmt)
         if (fmt != HTS_FMT_CSI) { // load linear index
             int j;
             if (bgzf_read(fp, &l->n, 4) != 4) return -1;
-            if (is_be) ed_swap_4p(&l->n);
+            if (is_be) ed_swap_8p(&l->n);
```

Comment:
I'm having a hard time convincing myself that this works — and at best since `l->n` went to 64 bits even on little endian it's subtly contingent on `l` just having been calloced.

Probably better to do the swapping on the data size that's in the file:
```
uint32_t x;
bgzf_read(fp, &x, 4);
if (is_be) ed_swap_4p(&x);
l->n = x;
```

Reply:
If it doesn't work on big-endian, then it doesn't on little-endian either for the same reason that not all the bytes get touched.  I will add a fix.

Reply:
I think actually it does work, it's just that it's super not obvious…

Reply:
…in any case, I guess it's this single line that caused the tabix test regression (as it became broken when `l->n` changed size in 1.10) that brought all this to a head in the distros' s390x testing. Which is pretty amusing :smile: